### DISCUSSION TITLE
How to set random seed for the model in YAML file

### DISCUSSION THREAD

**Discussion by PeterPirog** [ludwig-ai/ludwig#1532] [Q&A] [answered]
How to set random seed in YAML file to achieve repeatability of the model?

I found the code [https://ludwig-ai.github.io/ludwig-docs/api/LudwigModel/#create_model](https://ludwig-ai.github.io/ludwig-docs/api/LudwigModel/#create_model):
```
create_model(
  config,
  random_seed=42
)

```
but I don't know how to set it in YAML file.

Typical configuration in raw code is:
```
os.environ['PYTHONHASHSEED']=str(SEED)
os.environ['TF_CUDNN_DETERMINISTIC'] = '1'  # new flag present in tf 2.0+
random.seed(SEED)
np.random.seed(SEED)
tf.set_random_seed(SEED)
```

**Comment by tgaddair** [comment, marked as answer]
Hey @PeterPirog, good observation. Currently the random seed isn't configurable in the YAML, it is passed in as a command-line parameter, like:

```
ludwig train ... --random_seed 42
```

It can also be passed in to the `LudwigModel` programmatically:

```
model = LudwigModel(...)
model.train(..., random_seed=42)
```

It shouldn't be too difficult for us to add this to the config in the `training` section. What do you think @w4nderlust ?

**Reply by PeterPirog** [reply]
@tgaddair Thank You for the answer
